File: c_code/buffer.c
#include <stdio.h>
#include <string.h>

int main() {
    char buffer[8];
    int secret = 420;
    

    printf("Before buffer overflow: secret = %d\n", secret);
    strcpy(buffer, "This is a long string that will overflow the buffer");
    printf("After buffer overflow: secret = %d\n", secret);
    printf("Buffer content: %s\n", buffer);
    return 0;
}

File: rust_code/example_rust/src/bin/buffer.rs
fn main() {
    let buffer = [0u8; 8];
    let secret = 420;

    println!("Before buffer overflow: secret = {}", secret);
    println!("Buffer content: {:?}", buffer);
    println!("Value at index 20: {}", buffer[20]); // This will cause a panic due to out-of-bounds access
    println!("After buffer overflow: secret = {}", secret);
}